File: test/leetcode/m71/SimplifyPathTest.java
/*
 * Copyright (c) 2024 THGEEK.COM. All rights reserved.
 */

package leetcode.m71;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Test for {@link SimplifyPath}
 *
 * @author Tao Hong
 * @version 1.0
 * @since 2024/09/29 11:25
 */
class SimplifyPathTest {
    private SimplifyPath test;

    @BeforeEach
    void setUp() {
        test = new SimplifyPath();
    }

    @Test
    void test1() {
        String path = "/home/";
        String expected = "/home";
        assertEquals(expected, test.simplifyPath(path));
    }

    @Test
    void test2() {
        String path = "/home//foo/";
        String expected = "/home/foo";
        assertEquals(expected, test.simplifyPath(path));
    }

    @Test
    void test3() {
        String path = "/home/user/Documents/../Pictures";
        String expected = "/home/user/Pictures";
        assertEquals(expected, test.simplifyPath(path));
    }

    @Test
    void test4() {
        String path = "/../";
        String expected = "/";
        assertEquals(expected, test.simplifyPath(path));
    }

    @Test
    void test5() {
        String path = "/.../a/../b/c/../d/./";
        String expected = "/.../b/d";
        assertEquals(expected, test.simplifyPath(path));
    }
}

File: src/leetcode/m71/SimplifyPath_test.go
package m71

import "testing"

func Test_simplifyPath(t *testing.T) {
	tests := []struct {
		name string
		path string
		want string
	}{
		{"Test case 1", "/home/", "/home"},
		{"Test case 2", "/home//foo/", "/home/foo"},
		{"Test case 3", "/home/user/Documents/../Pictures", "/home/user/Pictures"},
		{"Test case 4", "/../", "/"},
		{"Test case 5", "/.../a/../b/c/../d/./", "/.../b/d"},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := simplifyPath(tt.path); got != tt.want {
				t.Errorf("simplifyPath() = %v, want %v", got, tt.want)
			}
		})
	}
}
